app/controllers/customers_controller.rb

```diff
@@ -0,0 +1,114 @@
+class CustomersController < ApplicationController
+  rescue_from ActiveRecord::RecordNotFound, with: :catch_not_found
+
+  before_action :set_customer, only: [:show, :edit, :update, :destroy]
+  layout 'customer_layout'
+
+  # GET /customers
+  # GET /customers.json
+  def index
+    @customers = Customer.all
+  end
+
+  # GET /customers/1
+  # GET /customers/1.json
+  def show
+  end
+
+  # GET /customers/new
+  def new
+    @customer = Customer.new
+  end
+
+  # GET /customers/1/edit
+  def edit
+  end
+
+  # POST /customers
+  # POST /customers.json
+#   def create
+#     @customer = Customer.new(customer_params)
+# @customer.save
+# flash.notice = "The customer record was created successfully."
+# redirect_to @customer
+# -----------------------------
+    # @customer = Customer.new(customer_params)
+    #
+    # respond_to do |format|
+    #   if @customer.save
+    #     format.html { redirect_to @customer, notice: 'Customer was successfully created.' }
+    #     format.json { render :show, status: :created, location: @customer }
+    #   else
+    #     format.html { render :new }
+    #     format.json { render json: @customer.errors, status: :unprocessable_entity }
+    #   end
+    # end
+    # -----------------------------
+
+    def create
+        @customer = Customer.new(customer_params)
+        if @customer.save
+          flash.notice = "The customer record was created successfully."
+          redirect_to @customer
+        else
+          flash.now.alert = @customer.errors.full_messages.to_sentence
+          render :new
+        end
+      end
+
+
+  end
+
+  # PATCH/PUT /customers/1
+  # PATCH/PUT /customers/1.json
+#   def update
+#     @customer.update(customer_params)
+# flash.notice = "The customer record was updated successfully."
+# redirect_to @customer
+    # respond_to do |format|
+    #   if @customer.update(customer_params)
+    #     format.html { redirect_to @customer, notice: 'Customer was successfully updated.' }
+    #     format.json { render :show, status: :ok, location: @customer }
+    #   else
+    #     format.html { render :edit }
+    #     format.json { render json: @customer.errors, status: :unprocessable_entity }
+    #   end
+    # end
+    # -------------------------------------
+    def update
+   if @customer.update(customer_params)
+     flash.notice = "The customer record was updated successfully."
+     redirect_to @customer
+   else
+     flash.now.alert = @customer.errors.full_messages.to_sentence
+     render :edit
+   end
+ end
+  end
```

Comment:
You have an extra `end` on line 87, which will cause the update block not to work.